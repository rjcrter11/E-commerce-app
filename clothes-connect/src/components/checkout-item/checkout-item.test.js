import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';

import { CheckoutItem } from './checkout-item.component';


describe('CheckoutItem Component', () => {
    let wrapper;
    let enzymeWrapper
    let mockClearItem;
    let mockAddItem;
    let mockRemoveItem;

    beforeEach(() => {
        mockClearItem = jest.fn();
        mockAddItem = jest.fn();
        mockRemoveItem = jest.fn();

        const mockProps = {
            cartItem: {
                imageUrl: 'www.testimg.com',
                price: 10,
                name: 'hats',
                quantity: 2
            },
            clearItem: mockClearItem,
            addItem: mockAddItem,
            removeItem: mockRemoveItem
        };

        wrapper = create(<CheckoutItem {...mockProps} />);
        enzymeWrapper = shallow(<CheckoutItem {...mockProps} />)
    });

    it('should render CheckoutItem component', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should call clearItem when remove button is clicked', () => {

        enzymeWrapper.find('[data-testid="remove-button"]').simulate('click')
        expect(mockClearItem).toHaveBeenCalled();
    })
    it('should call removeItem when left arrow is clicked', () => {
        enzymeWrapper.find('[data-testid="arrow-remove"]').simulate('click');
        expect(mockRemoveItem).toHaveBeenCalled();
    })
    it('should call addItem when right arrow is clicked', () => {
        enzymeWrapper.find('[data-testid="arrow-add"]').simulate('click');
        expect(mockAddItem).toHaveBeenCalled();
    })

})

