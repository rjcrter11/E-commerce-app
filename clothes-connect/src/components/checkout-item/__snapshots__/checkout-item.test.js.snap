// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutItem Component should render CheckoutItem component 1`] = `
<div
  className="checkout-item"
>
  <div
    className="image-container"
  >
    <img
      alt="item"
      src="www.testimg.com"
    />
  </div>
  <span
    className="name"
  >
    hats
  </span>
  <span
    className="quantity"
  >
    <div
      className="arrow"
      data-testid="arrow-remove"
      onClick={[Function]}
    >
      ❮
    </div>
    <span
      className="value"
    >
      2
    </span>
    <div
      className="arrow"
      data-testid="arrow-add"
      onClick={[Function]}
    >
      ❯
    </div>
  </span>
  <span
    className="price"
  >
     $
    10
     
  </span>
  <div
    className="remove-button"
    data-testid="remove-button"
    onClick={[Function]}
  >
    ✕
  </div>
</div>
`;
